cmd/smee/main.go

```diff
@@ -206,14 +208,27 @@ func main() {
 
 	// dhcp server
 	if cfg.dhcp.enabled {
-		listener, dh, err := cfg.dhcpListener(ctx, log)
+		dh, err := cfg.dhcpHandler(ctx, log)
 		if err != nil {
 			log.Error(err, "failed to create dhcp listener")
 			panic(fmt.Errorf("failed to create dhcp listener: %w", err))
 		}
 		log.Info("starting dhcp server", "bind_addr", cfg.dhcp.bindAddr)
 		g.Go(func() error {
-			return listener.ListenAndServe(ctx, dh)
+			bindAddr, err := netip.ParseAddrPort(cfg.dhcp.bindAddr)
+			if err != nil {
+				panic(fmt.Errorf("invalid tftp address for DHCP server: %w", err))
+			}
+			conn, err := server4.NewIPv4UDPConn("", net.UDPAddrFromAddrPort(bindAddr))
+			if err != nil {
+				panic(err)
+			}
+			defer func() {
+				_ = conn.Close()
+			}()
```

Comment:
Does this throw a linter error when defering `conn.Close()` directly? If so, can we just use `//nolint: ...`?

Reply:
its does. sure thing, will update.

Reply:
hmmm....actually i dont think the linting will complain. i must have just done it assuming the linter would complain. regardless, i've updated it.